### PR TITLE
[avmfritz] Fix for BigDecimal precision issues

### LINKED ISSUE #3244: [avmfritz] BigDecimal precision issues
Some users in the community observed an unexpected behavior sending commands to the `set_temp` channel. Values with .5 decimals works fine whereas integer values fails.

https://community.openhab.org/t/avm-fritzbox-resets-comet-dect-setpoint/31651
https://community.openhab.org/t/alexa-fritzbox-dect-thermostat/40046

## Current Behavior
Sending a decimal works fine (e.g. 16.5 is converted to `param=33`):
```   
2018-02-15 21:45:01.438 [DEBUG] [nding.avmfritz.handler.DeviceHandler] - Handle command '16.5' for channel set_temp
2018-02-15 21:45:01.445 [DEBUG] [ternal.hardware.FritzahaWebInterface] - GETting URL http://192.XXX.XXX.XXX/webservices/homeautoswitch.lua?ain=AAAAAAAAAA&switchcmd=sethkrtsoll&param=33&sid=818a862ffef3d5c3
2018-02-15 21:45:02.065 [DEBUG] [nal.hardware.FritzahaContentExchange] - HTTP response 200
2018-02-15 21:45:02.067 [DEBUG] [nal.hardware.FritzahaContentExchange] - response complete: 33
```

 Sending an integer fails (e.g. 20 is converted to `param=4E+1`):
```
2018-02-15 21:44:12.301 [DEBUG] [nding.avmfritz.handler.DeviceHandler] - Handle command '20' for channel set_temp
2018-02-15 21:44:12.318 [DEBUG] [ternal.hardware.FritzahaWebInterface] - GETting URL http://192.XXX.XXX.XXX/webservices/homeautoswitch.lua?ain=AAAAAAAAAA&switchcmd=sethkrtsoll&param=4E+1&sid=818a862ffef3d5c3
2018-02-15 21:44:13.360 [DEBUG] [nal.hardware.FritzahaContentExchange] - HTTP response 200
2018-02-15 21:44:13.365 [DEBUG] [nal.hardware.FritzahaContentExchange] - response complete:
```
## Expected Behavior

The integer commands are converted correctly e.g. 20 , converts to `param=40`.

## Context
FRITZ!Box accepts temperature values in .5 °C steps from 8 °C to 28 °C. Conversion uses the formula: `16 <= 8 °C, 17 = 8.5 °C, ... 56 >= 28 °C, 254 = ON, 253 = OFF`.

Conversion works like this (`TEMP_FACTOR = new BigDecimal("0.5")`):
https://github.com/openhab/openhab2-addons/blob/0444050ff9d662a48ed8ea934ad9bf4fbe67dc61/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java#L157-L166

## Possible Solution
I tackled it down to the well known issues with `BigDecimal` in Java.

I'm not sure how to solve this in a proper way. I found a solution which works for the binding. It replaces the division by a multiplication (`BIG_DECIMAL_TWO = new BigDecimal("2.0")`):
```
return BIG_DECIMAL_TWO.multiply(celsiusValue);
```

Any thoughts / advice on this?

### PR DESCRIPTION
**Pull request by cweitkamp**
See discussion in issue #3244 .

Closes #3244 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
         "nextchange" })
 public class HeatingModel {
     public static final BigDecimal TEMP_FACTOR = new BigDecimal("0.5");
+    public static final BigDecimal BIG_DECIMAL_TWO = new BigDecimal("2.0");
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java]: Isn't this caused by the number of significant numbers, ie would its also work if you replace in the old code "0.5" by e.g. "0.500"?

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java]: I'm not sure but I will test it and add feedback.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java]: I tested your proposal today but it results in the same error when we used "0.5".

Comment (martinvw) [reply addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java]: Pitty :-)

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/HeatingModel.java]: Yes, the `BigDecimal` seems to be a big mess. ;-)

**Pull request merged**